query/physicalplan/project.go

```diff
@@ -184,6 +184,8 @@ func projectionFromExpr(expr logicalplan.Expr) (columnProjection, error) {
 		return binaryExprProjection{
 			boolExpr: boolExpr,
 		}, nil
+	case *logicalplan.AverageExpr:
+		return &averageProjection{expr: e}, nil
```

Comment:
I'm not sure this is the right way to define this projection. In my mind, this projection is just a general division projection which intuitively fits into the switch case above: i.e. this is just a binary expression where the operator is the division operator. Not sure if our code is structured for this right now, but long term I think that's what we should do. This feels too specific to aggregations.

Reply:
Here's a changed implementation for making it less specific to the average case: https://github.com/polarsignals/frostdb/pull/302